components/accountPreferences/PasswordPreferences.tsx

```diff
@@ -0,0 +1,108 @@
+import { usePasswordChangeMutation } from "@/saleor/api";
+import React from "react";
+import { useForm } from "react-hook-form";
+
+export const PasswordPreferences: React.VFC<any> = ({}) => {
+  const [passwordChangeMutation] = usePasswordChangeMutation({});
+  const [successMessage, setSuccessMessage] = React.useState<String | null>(
+    null
+  );
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    setError,
+    clearErrors,
+  } = useForm();
+
+  const onPasswordPreferenceSubmit = handleSubmit(async (formData) => {
+    if (formData.newPassword !== formData.newPasswordRepeat) {
+      setError("error", { message: "Passwords have to match." });
+    } else {
+      const result = await passwordChangeMutation({
+        variables: {
+          newPassword: formData.newPassword,
+          oldPassword: formData.oldPassword,
+        },
+      });
+      const errors = result.data?.passwordChange?.errors || [];
+      if (errors.length > 0) {
+        errors.forEach((e) => setError("error", { message: e.message || "" }));
+        setTimeout(() => {
+          clearErrors();
+        }, 3000);
+        return;
+      } else if (result.data?.passwordChange?.user) {
+        setSuccessMessage("Password changed succesfully.");
+        setTimeout(() => {
+          setSuccessMessage(null);
+        }, 3000);
+      }
+    }
+  });
+  return (
+    <div className="mt-4 mb-4">
+      <h2 className="checkout-section-header-active mb-2">Change password</h2>
+      <form onSubmit={onPasswordPreferenceSubmit}>
+        <div className="grid grid-cols-12 gap-4 w-full">
+          <div className="col-span-full">
+            <label className="block pl-1 text-sm font-medium text-gray-700">
+              Old password
+            </label>
+            <input
+              className="px-4 py-2 rounded-md text-sm outline-none w-full"
+              type="password"
+              placeholder="Old password"
+              {...register("oldPassword", {
+                required: true,
+              })}
+            />
+          </div>
```

Comment:
```suggestion
            {!!errors.oldPassword && (
              <p className="mt-2 text-sm text-red-600">
                {errors.oldPassword.message}
              </p>
            )}
          </div>
```
And add errors to the rest of the fields accordingly.